From ca61524e05d785f0d14f40802c6a1a54a46b74bb Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Sat, 19 Jun 2021 01:50:26 +0200
Subject: [PATCH] CMakeLists.txt: add missing regex for zstd

Currently cmake provides an pc file with:
Libs.private:  -lbz2 -llzma -lZstd::Zstd -lz
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93b46a61..defc8745 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -299,6 +299,7 @@ foreach(LIB ${LIBS_PRIVATE})
 endforeach()
 string(REGEX REPLACE "-lBZip2::BZip2" "-lbz2" LIBS ${LIBS})
 string(REGEX REPLACE "-lLibLZMA::LibLZMA" "-llzma" LIBS ${LIBS})
+string(REGEX REPLACE "-lZstd::Zstd" "-lzstd" LIBS ${LIBS})
 string(REGEX REPLACE "-lOpenSSL::Crypto" "-lssl -lcrypto" LIBS ${LIBS})
 string(REGEX REPLACE "-lZLIB::ZLIB" "-lz" LIBS ${LIBS})
 string(REGEX REPLACE "-lGnuTLS::GnuTLS" "-lgnutls" LIBS ${LIBS})

From c4af9182efd9cbb127bb43486b55d9ddf4a521ca Mon Sep 17 00:00:00 2001
From: Thomas Klausner <tk@giga.or.at>
Date: Mon, 28 Jun 2021 11:52:15 +0200
Subject: [PATCH] Try having relative paths in pkg-config file.

Addresses #248
---
 CMakeLists.txt | 11 ++++++-----
 libzip.pc.in   |  7 +++----
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8611a405..93f89df1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -281,11 +281,12 @@ endif()
 
 
 # pkgconfig file
-set(prefix ${CMAKE_INSTALL_PREFIX})
-set(exec_prefix \${prefix})
-SET(bindir ${CMAKE_INSTALL_FULL_BINDIR})
-SET(libdir ${CMAKE_INSTALL_FULL_LIBDIR})
-SET(includedir ${CMAKE_INSTALL_FULL_INCLUDEDIR})
+file(RELATIVE_PATH pc_relative_bindir ${CMAKE_INSTALL_PREFIX} ${CMAKE_INSTALL_FULL_BINDIR})
+set(bindir "\${prefix}/${pc_relative_bindir}")
+file(RELATIVE_PATH pc_relative_libdir ${CMAKE_INSTALL_PREFIX} ${CMAKE_INSTALL_FULL_LIBDIR})
+set(libdir "\${prefix)/${pc_relative_libdir}")
+file(RELATIVE_PATH pc_relative_includedir ${CMAKE_INSTALL_PREFIX} ${CMAKE_INSTALL_FULL_INCLUDEDIR})
+set(includedir "\${prefix}/${pc_relative_includedir}")
 if(CMAKE_SYSTEM_NAME MATCHES BSD)
   set(PKG_CONFIG_RPATH "-Wl,-R\${libdir}")
 endif(CMAKE_SYSTEM_NAME MATCHES BSD)
diff --git a/libzip.pc.in b/libzip.pc.in
index 58b3dca7..d51b0ab6 100644
--- a/libzip.pc.in
+++ b/libzip.pc.in
@@ -1,10 +1,10 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
 bindir=@bindir@
 libdir=@libdir@
 includedir=@includedir@
 
-zipcmp=@bindir@/zipcmp
+zipcmp=${bindir}/zipcmp
 
 Name: libzip
 Description: library for handling zip archives
@@ -12,4 +12,3 @@ Version: @PROJECT_VERSION@
 Libs: @PKG_CONFIG_RPATH@ -L${libdir} -lzip
 Libs.private: @LIBS@
 Cflags: -I${includedir}
-

From 28951cbf403ea3a788f823377bbc171d1e1359ba Mon Sep 17 00:00:00 2001
From: Michael Heimpold <mhei@heimpold.de>
Date: Sat, 3 Jul 2021 00:04:14 +0200
Subject: [PATCH] Fix typo which prevents expansion of libdir in pkgconfig
 files

Signed-off-by: Michael Heimpold <mhei@heimpold.de>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93f89df1..2fb87165 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -284,7 +284,7 @@ endif()
 file(RELATIVE_PATH pc_relative_bindir ${CMAKE_INSTALL_PREFIX} ${CMAKE_INSTALL_FULL_BINDIR})
 set(bindir "\${prefix}/${pc_relative_bindir}")
 file(RELATIVE_PATH pc_relative_libdir ${CMAKE_INSTALL_PREFIX} ${CMAKE_INSTALL_FULL_LIBDIR})
-set(libdir "\${prefix)/${pc_relative_libdir}")
+set(libdir "\${prefix}/${pc_relative_libdir}")
 file(RELATIVE_PATH pc_relative_includedir ${CMAKE_INSTALL_PREFIX} ${CMAKE_INSTALL_FULL_INCLUDEDIR})
 set(includedir "\${prefix}/${pc_relative_includedir}")
 if(CMAKE_SYSTEM_NAME MATCHES BSD)
